### PR TITLE
icx: new module icx_facts

### PR DESCRIPTION
**Pull request by sushma-alethea**
SUMMARY
Running command to manage ICX device by Ansible automatically.

ISSUE TYPE
New Module Pull Request

COMPONENT NAME
ICX_7150

ADDITIONAL INFORMATION
Achieving of running command to manage ICX device by Ansible automatically.

### CODE REVIEW

**Comment by Akasurde** [comment]
Could you please squash all commits into one ? Thanks.

File: lib/ansible/modules/network/icx/icx_facts.py [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+author: "Ruckus Wireless (@Commscope)"
+short_description: Collect facts from remote Ruckus ICX 7000 series switches
+description:
+  - Collects a base set of device facts from a remote device that
+    is running ICX.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>). The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+notes:
+  - Tested against ICX 10.1
```
Comment (ganeshrn) [inline lib/ansible/modules/network/icx/icx_facts.py]: In the notes section, you can point to the icx platform guide https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/network/user_guide/platform_icx.rst

Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L107
Applicable for other icx modules as well.

File: test/units/modules/network/icx/fixtures/show_interfaces [hunk 2]
```diff
@@ -0,0 +1,2377 @@
...
+         0                   0                   0
+         1                   0                   0
+         2                   0                   0
+         3                   0                   0
+         4                   0                   0
+         5                   0                   0
+         6                   0                   0
+         7                   0                   0
+
+GigabitEthernet1/1/4 is down, line protocol is down
```
Comment (ganeshrn) [inline test/units/modules/network/icx/fixtures/show_interfaces]: Reduce the fixture output file size to reflect 2-3 interfaces in the output.

File: lib/ansible/modules/network/icx/icx_facts.py (hunk 1 above)
Comment (sushma-alethea) [reply lib/ansible/modules/network/icx/icx_facts.py]: The reason we have not added platform_icx.rst link, is currently only ssh connection is supported and that is like default. I think Junos supports other type of connections, hence they have added this link in notes. Hope this is fine.

Comment (sushma-alethea) [reply lib/ansible/modules/network/icx/icx_facts.py]: I think more appropriate will be when we have ansible-icx link similar to https://www.ansible.com/integrations/networks/juniper, then add that in notes.

Comment (ganeshrn) [reply lib/ansible/modules/network/icx/icx_facts.py]: Refer https://github.com/ansible/ansible/pull/59914#discussion_r313247935
Same doc guideline is followed for other modules.

**Pull request closed without merging**